srcalc/src/main/java/gov/va/med/srcalc/domain/variable/DiscreteNumericalVariable.java

```diff
@@ -0,0 +1,175 @@
+package gov.va.med.srcalc.domain.variable;
+
+import java.util.*;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Sort;
+import org.hibernate.annotations.SortType;
+
+/**
+ * A NumericalVariable that ultimately represents one of a finite, discrete
+ * set of values. This is mainly useful for a lab results (e.g., White Blood
+ * Count &lt;= 11.0x1000/mm^3 or &gt; 11.0x100/mm^3), but may be used for other
+ * numerical values such as Body Mass Index.
+ */
+@Entity
+@Table(name = "discrete_numerical_var")  // slightly abbreviate long table name
+public class DiscreteNumericalVariable extends NumericalVariable
+{
+    private SortedSet<Category> fCategories = new TreeSet<>();
+    
+    /**
+     * For reflection-based construction only. Business code should use
+     * {@link #DiscreteNumericalVariable(String, VariableGroup, String)}.
+     */
+    DiscreteNumericalVariable()
+    {
+    }
+    
+    public DiscreteNumericalVariable(
+            final String displayName,
+            final VariableGroup group,
+            final Set<Category> categories)
+    {
+        super(displayName, group);
+        fCategories = new TreeSet<>(categories);
+    }
+
+    @ElementCollection(fetch = FetchType.EAGER)  // eager-load due to close association
+    @Sort(type = SortType.NATURAL)
+    @CollectionTable(
+            name = "discrete_numerical_var_category",
+            joinColumns = @JoinColumn(name = "variable_id"))
+    public SortedSet<Category> getCategories()
+    {
+        return fCategories;
+    }
+
+    protected void setCategories(final SortedSet<Category> categories)
+    {
+        fCategories = categories;
+    }
+
+    @Override
+    public void accept(final VariableVisitor visitor) throws Exception
+    {
+        visitor.visitDiscreteNumerical(this);
+    }
+    
+    /**
+     * Returns the Category containing the given value, or null if none contains
+     * it.
+     * @return the containing Category or null
+     */
+    public Category getContainingCategory(final float value)
+    {
+        for (final Category c : getCategories())
```

Comment:
I don't like iterating for a search normally, but it's probably the best we can do here.